packages/main/src/Menu.js

```diff
@@ -0,0 +1,490 @@
+ * <li><code>Arrow Right</code>, <code>Space</code> or <code>Enter</code> - Opens a sub-menu if there are menu items nested
+ * in the currently selected menu item.</li>
+ * <li><code>Arrow Left</code> or <code>Escape</code> - Closes the currently opened sub-menu.</li>
+ * </ul>
+ * Note: if the text ditrection is set to Right-to-left (RTL), <code>Arrow Right</code> and <code>Arrow Left</code> functionality is swapped.
+ * <br>
+ *
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import @ui5/webcomponents/dist/Menu.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.Menu
+ * @extends UI5Element
+ * @tagname ui5-menu
+ * @appenddocs MenuItem
+ * @since 1.2.2
+ * @public
+ */
+class Menu extends UI5Element {
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get staticAreaStyles() {
+		return staticAreaMenuCss;
+	}
+
+	static get staticAreaTemplate() {
+		return staticAreaMenuTemplate;
+	}
+
+	static get dependencies() {
+		return [
+			ResponsivePopover,
+			Button,
+			List,
+			StandardListItem,
+			Icon,
+		];
+	}
+
+	static async onDefine() {
+		Menu.i18nBundle = await getI18nBundle("@ui5/webcomponents");
+	}
+
+	get itemsWithChildren() {
+		return !!this._currentItems.filter(item => item.item.items.length).length;
+	}
+
+	get itemsWithIcon() {
+		return !!this._currentItems.filter(item => item.item.icon !== "").length;
+	}
+
+	get isRtl() {
+		return this.effectiveDir === "rtl";
+	}
+
+	get placementType() {
+		const placement = this.isRtl ? "Left" : "Right";
+		return this._isSubMenu ? placement : "Bottom";
+	}
+
+	get verticalAlign() {
+		return this._isSubMenu ? "Top" : "Bottom";
+	}
+
+	get labelBack() {
+		return Menu.i18nBundle.getText(MENU_BACK_BUTTON_ARIA_LABEL);
+	}
+
+	get labelClose() {
+		return Menu.i18nBundle.getText(MENU_CLOSE_BUTTON_ARIA_LABEL);
+	}
+
+	get isPhone() {
+		return isPhone();
+	}
+
+	get isSubMenuOpened() {
+		return !!this._parentMenuItem;
+	}
+
+	get menuTitlePhone() {
+		return this.isSubMenuOpened ? this._parentMenuItem.text : this.title;
+	}
+
+	onBeforeRendering() {
+		!isPhone() && this._prepareCurrentItems(this.items);
+
+		const itemsWithChildren = this.itemsWithChildren;
+		const itemsWithIcon = this.itemsWithIcon;
+
+		this._currentItems.forEach(item => {
+			item.item._siblingsWithChildren = itemsWithChildren;
+			item.item._siblingsWithIcon = itemsWithIcon;
+		});
+	}
+
+	/**
+	 * Shows the Menu near the opener element.
+	 * @param {HTMLElement} opener the element that the popover is shown at
+	 * @public
+	 */
+	async showAt(opener) {
+		if (isPhone()) {
+			this._prepareCurrentItems(this.items);
+			this._parentItemsStack = [];
+		}
+		if (!this._isSubMenu) {
+			this._parentMenuItem = undefined;
+		}
+		(await this._getPopover()).showAt(opener);
+	}
+
+	/**
+	 * Closes the Menu.
+	 * @public
+	 */
+	close() {
+		if (Object.keys(this._popover).length) {
```

Comment:
Isn't  "if (this._popover)" enough, maybe I miss something?

Reply:
Nope. _popover is defined as Object type, and when it is not instantiated yet, it is still Object, but empty one, i.e. there are no properties, so length is 0. When it is  instantiated, it is also object, but with properties. So in my opinion the length of the properties is legit way to check if there is something in this object.